substrate/frame/offences/src/migration.rs

```diff
@@ -106,12 +106,7 @@ pub fn remove_deferred_storage<T: Config>() -> Weight {
 	let deferred = <DeferredOffences<T>>::take();
 	log::info!(target: LOG_TARGET, "have {} deferred offences, applying.", deferred.len());
 	for (offences, perbill, session) in deferred.iter() {
-		let consumed = T::OnOffenceHandler::on_offence(
-			offences,
-			perbill,
-			*session,
-			DisableStrategy::WhenSlashed,
-		);
```

Comment:
Is this okay?

Reply:
On a second thought, this should be fine. We might call `on_offence` in the migration of deferred offences even for 0% slashes, but that is excepted according to the new implementation.

Reply:
I agree @gpestana but in this case the function signature is changed (removing `DisableStrategy::WhenSlashed`). If I don't modify the migration it won't compile. There are no modifications related to storage in the code so I believe this should be good?

Reply:
Actually `remove_deferred_storage` is used only in [`should_resubmit_deferred_offences`](https://github.com/paritytech/polkadot-sdk/blob/bbb6631641f9adba30c0ee6f4d11023a424dd362/substrate/frame/offences/src/migration.rs#L146) which is a unit test.

So this is not an issue.